### PR TITLE
Add different strategies examples for kustomize patches fields

### PR DESCRIPTION
**Pull request by Kissy**
Improved the documentation of kustomize `patches` field that support a lot of different writing and features.

Added a working example that can be copy & pasted.
Explained some specificities like a label containing a forward slash that need to be escaped which is always the case for Kubernetes recommended labels.

Signed-off-by: Guillaume Le Biller <glebiller@Traveldoo.com>

### CODE REVIEW

File: site/content/en/references/kustomize/kustomization/patches/_index.md [hunk 1]
```diff
@@ -55,122 +55,213 @@ The `name` and `namespace` fields of the patch target selector are
 automatically anchored regular expressions. This means that the value `myapp`
 is equivalent to `^myapp$`. 
 
-Consider the following `deployment.yaml` common for both the examples:
+With patches it is possible to override the kind or name of the resource it is editing.
+By default, the patch will leave the kind and name of the resource untouched. For example:
```
Comment (natasha41575) [inline site/content/en/references/kustomize/kustomization/patches/_index.md:59]: Let's change the ordering of things for clarity. Please change this to:

With patches it is possible to override the kind or name of the resource it is editing with the options `allowNameChange` and `allowKindChange`. For example:

< example >

By default, these fields are false and the patch will leave the kind and name of the resource untouched.

File: site/content/en/references/kustomize/kustomization/patches/_index.md [hunk 2]
```diff
@@ -17,12 +17,12 @@ The `patches` field contains a list of patches to be applied in the order they a
...
 - be either a file, or an inline string
 - target a single resource or multiple resources
 
-The patch target selects resources by group, version, kind, name, namespace, labelSelector and
-annotationSelector. Any resource which matches all the **specified** fields has the patch applied
+The patch target selects resources by `group`, `version`, `kind`, `name`, `namespace`, `labelSelector` and
+`annotationSelector`. Any resource which matches all the **specified** fields has the patch applied
 to it (regular expressions). 
 
 A patch can refer to a resource by any of its previous names or kinds.
```
Comment (natasha41575) [inline site/content/en/references/kustomize/kustomization/patches/_index.md:28]: Let's move this paragraph to right above `##Examples`

**Review by natasha41575 — COMMENTED** [review]
Thank you for the PR! This is a massive improvement. Couple of suggestions to improve it a little more

File: site/content/en/references/kustomize/kustomization/patches/_index.md (hunk 2 above)
Comment (Kissy) [reply site/content/en/references/kustomize/kustomization/patches/_index.md]: Thank you for both changes, it make a lot more sense to group everything that way :)

File: site/content/en/references/kustomize/kustomization/patches/_index.md [hunk 3]
```diff
@@ -55,122 +51,218 @@ The `name` and `namespace` fields of the patch target selector are
...
-- target:
-    kind: Deployment
-    name: the-deployment
-  path: patch.json
+  - path: add-label.patch.json
+    target:
+      group: apps
+      version: v1
+      kind: Deployment
+  - patch: fix-version.patch.yaml
```
Comment (natasha41575) [inline site/content/en/references/kustomize/kustomization/patches/_index.md:218]: just noticed this - is this `patch` here supposed to be `path`?

File: site/content/en/references/kustomize/kustomization/patches/_index.md [hunk 4]
```diff
@@ -55,122 +51,218 @@ The `name` and `namespace` fields of the patch target selector are
...
 
 ```yaml
 # kustomization.yaml
 resources:
 - deployment.yaml
-
 patches:
-- path: patch.yaml
+  - path: add-label.patch.yaml
+  - patch: fix-version.patch.yaml
```
Comment (natasha41575) [inline site/content/en/references/kustomize/kustomization/patches/_index.md:173]: `patch` -> `path`?

File: site/content/en/references/kustomize/kustomization/patches/_index.md [hunk 5]
```diff
@@ -55,122 +51,218 @@ The `name` and `namespace` fields of the patch target selector are
...
+### Patch using Path Strategic Merge
 
 ```yaml
 # kustomization.yaml
 resources:
 - deployment.yaml
-
 patches:
-- path: patch.yaml
+  - path: add-label.patch.yaml
```
Comment (natasha41575) [inline site/content/en/references/kustomize/kustomization/patches/_index.md:172]: Please add a comment explaining what happens if no target is specified

File: site/content/en/references/kustomize/kustomization/patches/_index.md (hunk 3 above)
Comment (Kissy) [reply site/content/en/references/kustomize/kustomization/patches/_index.md:218]: Totally; I indeed modified it when trying locally. Let me double check all examples are working! Thank you

**Comment by Kissy** [comment]
Fix patch -> path and re-run a `kustomize build` on all examples to verify the syntax.
Added a note about the behavior when the target field is omitted.

**Pull request merged**